rune/libcontainer/container_linux.go

```diff
@@ -530,6 +530,13 @@ func (c *linuxContainer) commandTemplate(p *Process, childInitPipe *os.File, chi
 			cmd.Env = append(cmd.Env,
 				fmt.Sprintf("_LIBCONTAINER_AGENTPIPE=%d", stdioFdCount+len(cmd.ExtraFiles)-1))
 		}
+
+		if c.config.Enclave.Path != "" {
+			cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_PATH=" + string(c.config.Enclave.Path))
+		}
+		if c.config.Enclave.Signer != "server" {
+			cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_ROOTFS=" + string(c.config.Rootfs))
+		}
```

Comment:
You should handle this in a better way. In server signer mode, _LIBCONTAINER_PAL_PATH should be constructed with a bundle rootfs prefix: `cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_PATH=" + string(c.config.Enclave.Path))`, and in client signer mode, _LIBCONTAINER_PAL_PATH should be constructed as `cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_PATH=" + c.config.Rootfs + os_path_separator + c.config.Enclave.Path)`. You don't really need an extra variable _LIBCONTAINER_PAL_ROOTFS.

Reply:
> 
> 
> You should handle this in a better way. In server signer mode, _LIBCONTAINER_PAL_PATH should be constructed with a bundle rootfs prefix: `cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_PATH=" + string(c.config.Enclave.Path))`, and in client signer mode, _LIBCONTAINER_PAL_PATH should be constructed as `cmd.Env = append(cmd.Env, "_LIBCONTAINER_PAL_PATH=" + c.config.Rootfs + os_path_separator + c.config.Enclave.Path)`. You don't really need an extra variable _LIBCONTAINER_PAL_ROOTFS.

This method doesn't work, The extra path `/lib64` should be used by dlopen().